### ISSUE TITLE
[BUG] Cannot edit content of a static placeholder on a regular page from a django app if it's set on the homepage of the website

### ISSUE THREAD

**Issue by corentinbettiol** [django-cms/django-cms#7544]
## Description

We have a site with an interactive map that we've created in a regular django app. I need to let our customer add a small text on the top of the page. I though that adding a static placeholder will do the job fine. I tried. I failed.

## Steps to reproduce

> Main urls.py file: `website/urls.py`
```py
# [...]
urlpatterns += i18n_patterns(
    path(
        "",
        include(("website.apps.catalog.urls", "catalog"), namespace="catalog"),
    ),
    # CMS - should be the last because it catches all urls
    re_path(r"^", include("cms.urls")),
)
# [...]
```

> Our app urls.py file: `website/apps/catalog/urls.py`
```py
# [...]
urlpatterns = [
    path(
        _("catalog/<str:type_of_structure>/<slug:slug>/"),
        views.PlaceDetailView.as_view(),
        name="detail_view",
    ),
    path(
        "",
        views.ResultMapView.as_view(),
        name="result_map_view",
    ),
]
# [...]
```

Our `ResultMapView` then loads some objects and uses the `result_map_view.html` template.

> `result_map_view.html`:
```jinja
{# ... #}
{% block header %}
  {% include "kapt_templates/partials/header.html" %}
  {% static_placeholder "Banner result map view" %}
{% endblock header %}
{# ... #}
```

We created a new homepage on the site (since we need some other djangocms pages, and djangocms *needs* a homepage).

After a refresh, the Structure button is activated on the interactive map (that is also the homepage, see the `urls.py` files above).

## Actual behaviour

The structure button is activated (as it should be), but when we click on it, it displays an empty structure panel.

## Expected behaviour

The structure button should still be activated, but displaying the structure menu should also display the static placeholder entry.

## Screenshots

![image](https://user-images.githubusercontent.com/45763865/234798171-34c336cc-2706-48b6-afa2-39d48c66afc8.png)

## Additional information (CMS/Python/Django versions)

Python 3.10-ish, Django 3.2, Django CMS 3.11.3.

## Do you want to help fix this issue?

* [x] Yes, I want to help fix this issue and I will join #workgroup-pr-review on [Slack](https://www.django-cms.org/slack) to confirm with the community that a PR is welcome. 
* [ ] No, I only want to report the issue.

Can you give me some insights on what's going wrong, so I can craft a PR? I think it might be that because the django app is on the homepage (empty path in urls.py) and a cms page exist (on the same path, since it's the homepage), the static placeholder fail to load everything it should, but I could be wrong, and I'm not familiar enough with django-cms's code to search efficiently.

**Comment by corentinbettiol** [comment]
I may be able to fix this, but I don't know where to start. Could someone show me where to start?

**Comment by fsbraun** [comment]
@corentinbettiol I am not sure why that happens. Probably you need to do some code reading. Do I understand correctly that it happens on a page which is resolved to a non-CMS page? Does it also have to be the root page?

A workaround might be to add a `PlaceholderField` to a model of the non-cms app and show that placeholder_field on its page? (https://docs.django-cms.org/en/latest/how_to/placeholders.html)

**Comment by corentinbettiol** [comment]
> Do I understand correctly that it happens on a page which is resolved to a non-CMS page? Does it also have to be the root page?

Yeah that's it. The root page of the site is an interactive map, which is defined in `urls.py`:

```py
urlpatterns += i18n_patterns(
    # [other urls]
    path(
        "",
        include(("website.apps.catalog.urls", "catalog"), namespace="catalog"),
    ),
    # [other urls]
    re_path(r"^", include("cms.urls")),
)
```

So there *is* a cms page that's defined as the root page (because we need a root page? I don't really know), but what's displayed if the page from `website.apps.catalog.urls`.

Here's the cms "root" page (that we can't access):

![image](https://github.com/django-cms/django-cms/assets/45763865/b4bc7449-946f-47fb-887b-dfc1f1ef3382)

**Comment by fsbraun** [comment]
@corentinbettiol I think there is no way around this. Two work arounds come to mind:

1. Replace the static placeholder by a model with a `HTMLField`. You can make out frontend editable so that double clicking opens the model's admin 
2. Add the static placeholder to a different page and edit and publish it from there. 
3. Make the page a CMS page and create a plugin or template tag that provides the necessary context for your map. 

Well, the work arounds come to mind...
